src/app/confirm-call-destination/_components/ActionButton.tsx

```diff
@@ -28,31 +28,73 @@ const useActionButton = () =>{
 }
 
 const useSendMessage = (endpoint:string) =>{
+  const [isLoading,setIsLoading] = useState(false)
+  const router = useRouter()
   const sendMessage: React.ComponentProps<typeof ActionButtons>['onConfirm'] = async(e) => {
+    setIsLoading(true)
     e.preventDefault();
     const response = await fetch(`${ENDPOINT_BASE_PATH}/${endpoint}`, {
       method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
     })
-
-    //別の画面でエラーメッセージとしてアラートで表示する予定のためreturnしています。
-    if (!response.ok) {
-      return { error: 'Failed to send message', status: response.status }
-    }
-    //レスポンスが200なのか401なのかによって呼び出し完了画面と呼び出し失敗画面のどちらに遷移するかを決めるためreturnしています。
-    const data = await response.json()
+    setIsLoading(false)
+    const data = response.status
     return data
   }
- return sendMessage
+ return {sendMessage,isLoading}
```

Comment:
ここの名前は `(handle or on)Confirm` にしたいです

```suggestion
 return { handleConfirm: sendMessage, isLoading }
```

Reply:
sendMessageを`(handle or on)Confirm`にrenameするでも良しです